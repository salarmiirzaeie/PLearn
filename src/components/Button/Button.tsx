// PrimaryButton.tsx
import React from 'react';
import {Text, TouchableOpacity} from 'react-native';

import {ButtonContainer, ButtonText} from '../../configs/styled-components';

interface PrimaryButtonProps {
  onPress: () => void;
  title: string;
  color?: string;
  textColor?: string;
}

const Button: React.FC<PrimaryButtonProps> = ({
  onPress,
  title,
  color,
  textColor,
}) => (
  <ButtonContainer color={color} onPress={onPress}>
    <ButtonText color={textColor}>{title}</ButtonText>
  </ButtonContainer>
);

export default Button;
