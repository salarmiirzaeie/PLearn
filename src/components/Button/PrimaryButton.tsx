// PrimaryButton.tsx
import React from 'react';
import {TouchableOpacity, Text} from 'react-native';
import styled from 'styled-components/native';
interface ButtonContainerProps {
  color?: string; // پارامتر رنگ
}
const ButtonContainer = styled(TouchableOpacity)<ButtonContainerProps>`
  background-color: ${props => props.color || '#004CFF'};
  padding: 10px 20px;
  border-radius: 10px;
`;

const ButtonText = styled(Text)`
  color: #fff;
  font-size: 16px;
  text-align: center;
`;

interface PrimaryButtonProps {
  onPress: () => void;
  title: string;
  color?: string;
}

const PrimaryButton: React.FC<PrimaryButtonProps> = ({
  onPress,
  title,
  color,
}) => (
  <ButtonContainer color={color} onPress={onPress}>
    <ButtonText>{title}</ButtonText>
  </ButtonContainer>
);

export default PrimaryButton;
