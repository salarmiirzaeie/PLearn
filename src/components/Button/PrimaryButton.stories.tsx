// stories/PrimaryButton.stories.tsx
import type {Meta, StoryObj} from '@storybook/react';

import PrimaryButton from './PrimaryButton';

const meta = {
  title: 'components/PrimaryButton',
  component: PrimaryButton,
} satisfies Meta<typeof PrimaryButton>;

export default meta;

type Story = StoryObj<typeof meta>;

export const Basic: Story = {
  args: {
    title: 'Hello World',
    color: '#004CFF',
  },
};
