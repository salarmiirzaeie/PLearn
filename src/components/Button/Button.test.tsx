// PrimaryButton.test.tsx
import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import Button from './Button';

test('PrimaryButton renders correctly', () => {
  const tree = renderer
    .create(<Button onPress={() => {}} title="Test Button" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
