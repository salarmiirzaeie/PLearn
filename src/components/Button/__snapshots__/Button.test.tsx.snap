// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "#65CAFF",
      "borderBottomLeftRadius": 10,
      "borderBottomRightRadius": 10,
      "borderTopLeftRadius": 10,
      "borderTopRightRadius": 10,
      "opacity": 1,
      "paddingBottom": 10,
      "paddingLeft": 20,
      "paddingRight": 20,
      "paddingTop": 10,
    }
  }
>
  <Text
    style={
      {
        "color": "#fff",
        "fontSize": 16,
        "textAlign": "center",
      }
    }
  >
    Test Button
  </Text>
</View>
`;
